staticd/static_srv6.c

```diff
@@ -253,6 +273,654 @@ void static_zebra_request_srv6_sids(void)
+{
+	XFREE(MTYPE_STATIC_SRV6_NEIGH, neigh);
+}
+
+/*
+ * Initialize the neighbor cache
+ */
+void static_srv6_neigh_cache_init(void)
+{
+	if (neigh_cache)
+		return;
+
+	neigh_cache = XCALLOC(MTYPE_STATIC_SRV6_NEIGH_CACHE,
+			      sizeof(struct static_srv6_neigh_cache));
+
+	static_srv6_neigh_table_init(&neigh_cache->neigh_table);
+	neigh_cache->resolve_sids_cnt = 0;
+	neigh_cache->registered = false;
+}
+
+/*
+ * Clean up the neighbor cache
+ */
+void static_srv6_neigh_cache_cleanup(void)
+{
+	struct static_srv6_if_neigh *ifn;
+	struct static_srv6_neigh *neigh;
+	struct static_srv6_neigh *next;
+
+	if (!neigh_cache)
+		return;
+
+	/* Unregister from neighbor notifications if needed */
+	if (neigh_cache->registered)
+		static_srv6_neigh_unregister_if_needed();
+
+	/* Clean up hash table - free all interface neighbors */
+	while ((ifn = static_srv6_neigh_table_pop(&neigh_cache->neigh_table)) != NULL) {
+		neigh = ifn->neighbors;
+		while (neigh) {
+			next = neigh->next;
+			static_srv6_neigh_free(neigh);
+			neigh = next;
+		}
+		XFREE(MTYPE_STATIC_SRV6_IF_NEIGH, ifn);
+	}
+	static_srv6_neigh_table_fini(&neigh_cache->neigh_table);
+
+	XFREE(MTYPE_STATIC_SRV6_NEIGH_CACHE, neigh_cache);
+	neigh_cache = NULL;
+}
+
+/*
+ * Check if a neighbor state is usable for SID resolution
+ * Returns true if the neighbor can be used
+ */
+static bool static_srv6_neigh_state_is_usable(uint32_t ndm_state)
+{
+	/* States that indicate the neighbor is reachable/usable */
+	if (ndm_state &
+	    (ZEBRA_NEIGH_STATE_REACHABLE | ZEBRA_NEIGH_STATE_PERMANENT | ZEBRA_NEIGH_STATE_NOARP))
+		return true;
+
+	/* States that indicate problems */
+	if (ndm_state & (ZEBRA_NEIGH_STATE_FAILED | ZEBRA_NEIGH_STATE_INCOMPLETE))
+		return false;
+
+	/*
+	 * For other states (STALE, DELAY, PROBE), be conservative
+	 * and consider them usable - the kernel will handle revalidation
+	 */
+	if (ndm_state &
+	    (ZEBRA_NEIGH_STATE_STALE | ZEBRA_NEIGH_STATE_DELAY | ZEBRA_NEIGH_STATE_PROBE))
+		return true;
+
+	/* If no state bits set, assume not usable */
+	return false;
+}
+
+/**
+ * Refresh SIDs when a neighbor is added
+ * Try to resolve and install previously unresolved SIDs
+ */
+void static_srv6_refresh_sids_on_neigh_change(struct interface *ifp, struct in6_addr *nexthop,
+					      bool is_add)
+{
+	struct listnode *node;
+	struct static_srv6_sid *sid;
+	int processed = 0;
+
+	if (!srv6_sids) {
+		DEBUGD(&static_dbg_srv6, "%s: No SIDs available for refresh", __func__);
+		return;
+	}
+
+	DEBUGD(&static_dbg_srv6, "%s: Refreshing SIDs after neighbor %s to interface %u", __func__,
+	       is_add ? "added" : "removed", ifp->ifindex);
+
+	for (ALL_LIST_ELEMENTS_RO(srv6_sids, node, sid)) {
+		/* Skip SIDs that don't require nexthop resolution */
+		if (!static_srv6_sid_needs_resolution(sid))
+			continue;
+
+		/* Check if this SID is for this interface */
+		if (!strmatch(sid->attributes.ifname, ifp->name))
+			continue;
```

Comment:
// Line 405: Name comparison
if (!strmatch(sid->attributes.ifname, ifp->name))

// vs Line 674: Index-based lookup
struct static_srv6_if_neigh lookup_key = { .ifindex = ifp->ifindex };

Interface "eth0" gets renamed to "eth1" via `ip link` The ifindex may  stay the same (e.g., 3), but the SID configuration still may reference the old name "eth0". When `static_srv6_refresh_sids_on_neigh_change()` uses `strmatch(sid->attributes.ifname, ifp->name)`, it might not find the SID because names don't match, even though it's the same interface. 

Would it be nice to to use ifindex for matching instead of names?